falcon/routing/compiled.py

```diff
@@ -362,4 +358,4 @@ def conflicts_with(self, segment):
                 #   /foo/all
                 return True
```

Comment:
I think we could support this by sorting the routing logic so it always tests static routes first, regardless of what order add_route was called in. I can take a stab at implementing that once your patch is merged.